drivers: flash: stm32: get write block size from device tree

erwango opened the pull request:
In flash driver init, wrtie_block_size could be initialized
with FLASH_WRITE_BLOCK_SIZE which is generated from device
tree parsing (dts property: "write-block-size")
Solves #5305

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

pfalcon commented:
@nashif, @galak : Please merge.

ydamigos reviewed (approved):
LGTM, just a typo in commits body.
wrtie_block_size -> write_block_size

galak commented on drivers/flash/flash_stm32.c:
```diff
@@ -224,13 +224,7 @@ static const struct flash_driver_api flash_stm32_api = {
 #ifdef CONFIG_FLASH_PAGE_LAYOUT
 	.page_layout = flash_stm32_page_layout,
 #endif
-#if defined(CONFIG_SOC_SERIES_STM32F0X)
```
Should this be #if defined(FLASH_WRITE_BLOCK_SIZE) instead?

superna9999 replied:
Yes it should !

galak reviewed (changes requested):
Question about how the ifdef is defined/used if we get this from DT.

erwango committed:
drivers: flash: stm32: get write block size from device tree

In flash driver init, write_block_size could be initialized
with FLASH_WRITE_BLOCK_SIZE which is generated from device
tree parsing (dts property: "write-block-size")
Fixes #5305
If not defined, generate a compilation error.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

mbolivar reviewed (commented):
No objections; agree with suggestion to change to defined(FLASH_WRITE_BLOCK_SIZE).
